pkg/controller/deployment/deployment_controller.go

```diff
@@ -426,6 +426,11 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	}
 
 	d := obj.(*extensions.Deployment)
+	everything := unversioned.LabelSelector{}
+	if reflect.DeepEqual(d.Spec.Selector, &everything) {
+		dc.eventRecorder.Eventf(d, api.EventTypeWarning, "SelectingAll", "This controller is selecting all pods. Skipping sync.")
```

Comment:
s/controller/Deployment/

I doubt "skipping sync" is very meaningful to users. How about "A non-empty selector is required."